Ramos_HwAssignments/Ramos_Wk4_WeatherApp/Weather.tsx

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react';
+import { NavigationContainer, useNavigation, useRoute, RouteProp, StackActions } from '@react-navigation/native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import { Text, View, SafeAreaView, TextInput, Alert, FlatList, TouchableOpacity } from 'react-native';
+import { StyleSheet } from 'react-native';
+import { StackParamList } from './App';
+import { useFetch } from './Functions/useFetch';
+import { PhotoItem } from './PhotoItem';
+
+type WeatherNavigationProp = StackNavigationProp<StackParamList, 'Weather'>;
+
+export const Weather = () => {
+  const navigation = useNavigation<WeatherNavigationProp>();
+
+  const {data, loading, error} = useFetch('http://api.weatherapi.com/v1/current.json?q=02893&key=2c1c6cbbec964d26907230624231608');
+
+  const [location, setLocation] = useState('');
+  const [currentWeather, setCurrentWeather] =  useState('');
+
+  useEffect(() => {
+    setLocation(data?.location);
```

Comment:
You don't need `useEffect` or `useState` here because `useFetch` is already setting state and causes a re-render.